src/cursor-doc/paredit.ts

```diff
@@ -138,6 +138,90 @@ export function backwardListRange(doc: EditableDocument, start: number = doc.sel
+ * When there is a newline before the end of the current form either:
+ *  - Return the end of the nearest form to the right of the cursor location if one exists
+ *  - Returns the newline's offset if no form exists
+ *
+ * This function's output range is needed to implement features similar to paredit's
+ * killRight or smartparens' sp-kill-hybrid-sexp.
+ *
+ * @param doc
+ * @param offset
+ * @param goPastWhitespace
+ * @returns [number, number]
+ */
+export function forwardHybridSexpRange(doc: EditableDocument, offset = Math.max(doc.selection.anchor, doc.selection.active), goPastWhitespace = false): [number, number] {
+    let cursor = doc.getTokenCursor(offset);
+    if (cursor.getToken().type === 'open') {
+        return forwardSexpRange(doc);
+    } else if (cursor.getToken().type === 'close') {
+        return [offset, offset];
+    }
+
+    const currentLineText = doc.model.getLineText(cursor.line);
+    const lineStart = doc.model.getOffsetForLine(cursor.line);
+    const currentLineNewlineOffset = lineStart + currentLineText.length;
+    const remainderLineText = doc.model.getText(offset, currentLineNewlineOffset + 1);
+
+    cursor.forwardList(); // move to the end of the current form
+    const currentFormEndToken = cursor.getToken();
+    // when we've advanced the cursor but start is behind us then go to the end
+    // happens when in a clojure comment i.e:  ;; ----
+    let cursorOffsetEnd = cursor.offsetStart <= offset ? cursor.offsetEnd : cursor.offsetStart;
+    const text = doc.model.getText(offset, cursorOffsetEnd);
+    let hasNewline = text.indexOf("\n") > -1;
+    let end = cursorOffsetEnd;
+
+    // Want the min of closing token or newline
+    // After moving forward, the cursor is not yet at the end of the current line,
+    // and it is not a close token. So we include the newline
+    // because what forms are here extend beyond the end of the current line
+    if (currentLineNewlineOffset > cursor.offsetEnd && currentFormEndToken.type != 'close') {
+        hasNewline = true;
+        end = currentLineNewlineOffset;
+    }
+
+
+    // handle weird CRLF case first otherwise we'll not advance far enough
+    if (remainderLineText === '\n' && '\n' === doc.model.getText(currentLineNewlineOffset, currentLineNewlineOffset + 2)) {
+        end = currentLineNewlineOffset + 2;
+    } else if (remainderLineText === '' || remainderLineText === '\n') {
+        end = currentLineNewlineOffset + 1;
+    } else if (hasNewline) {
+        // Try to find the first open token to the right of the document's cursor location if any
+        let nearestOpenTokenOffset = -1;
+
+        // Start at the newline.
+        // Work backwards to find the smallest open token offset
+        // greater than the document's cursor location if any
+        cursor = doc.getTokenCursor(currentLineNewlineOffset);
+        while(cursor.offsetStart > offset) {
+            // backwardSexp() returns false when it gets to an open token type
```

Comment:
I think this comment is not necessary now that you have understood this. It is a fundamental behavior of the cursor movements. It's documented on `backwardsSexp()`.